contrib/vagrant/Makefile

```diff
@@ -0,0 +1,509 @@
+
+VALIDATE_MANIFEST = \
+	if [ -f "$(1)" ] && echo "$(2)" | diff "$(1)" - 2>&1 >/dev/null; then \
+	    rm -f "$(1)"; \
+	fi
+
+# ===----------------------------------------------------------------------===
+
+PKGS            :=
+CACHE_DIR       := cache
+HOST_CACHE_DIR  := $(CACHE_DIR)
+GUEST_CACHE_DIR := /vagrant/$(CACHE_DIR)
+
+SOURCEFORGE_MIRROR := downloads.sourceforge.net
+
+PKG             := osslsigncode-1.7.1.tar.gz
+$(PKG)_CHECKSUM := 3b53ccf0243cd4a6a731a898bfdba95c3d22342e7e64f9e9a4a2bce5f74ac809943f570ed92c9077101a178f82f49ece1d1c4bfee1f288ea61d99fb36f6a243e
+$(PKG)_URL      := http://$(SOURCEFORGE_MIRROR)/project/osslsigncode/osslsigncode/$(PKG)
+PKGS            += $(PKG)
+
+PKG             := osslsigncode-Backports-to-1.7.1.patch
+$(PKG)_CHECKSUM := 182127638206492a609b4f909465b702f7fa89b3cd69f08f86a2efde86dbe597081fb0fd862560fac89e85baf74976c9bcb6b56d17f0e9015ea0c7e33cae1e3a
+$(PKG)_URL      := http://bitcoincore.org/cfields/osslsigncode-Backports-to-1.7.1.patch
+PKGS            += $(PKG)
+
+# ===----------------------------------------------------------------------===
+
+VAGRANT_UP = \
+	vagrant up
+
+VAGRANT_HALT = \
+	vagrant halt
+
+VAGRANT_RELOAD = \
+	vagrant reload
+
+VAGRANT_DESTROY = \
+	vagrant destroy --force
+
+# ===----------------------------------------------------------------------===
+
+.PHONY: all
+all: linux osx win
+
+.PHONY: mostlyclean
+mostlyclean:
+
+.PHONY: clean
+clean: mostlyclean
+	-rm -rf box
+	-rm -rf output
+	-rm -f .stamp-*
+	-$(call VAGRANT_DESTROY)
+	-rm -rf .vagrant
+
+.PHONY: distclean
+distclean: clean
+	-rm -rf cache
+	-rm -rf output-gitian-host-xenial64
+	-rm -rf packer_cache
+
+.PHONY: maintainer-clean
+maintainer-clean: distclean
+	@echo 'This command is intended for maintainers to use; it'
+	@echo 'deletes files that may need special tools to rebuild.'
+
+.PHONY: dist
+dist: all
+
+# ===----------------------------------------------------------------------===
+
+.PHONY: check-requirements
+define CHECK_REQUIREMENT
+	@if ! $(1) --help &>/dev/null; then \
+	    if ! which $(1) &>/dev/null; then \
+	        echo; \
+	        echo 'Missing requirement: $(1)'; \
+	        echo; \
+	        echo 'Please have a look at "README" to ensure'; \
+	        echo 'that your system meets all requirements.'; \
+	        echo; \
+	        exit 1; \
+	    fi; \
+	fi
+
+endef
+define CHECK_REQUIREMENT_VERSION
+	@if ! $(1) --version | head -1 | grep ' \($(2)\)$$' >/dev/null; then \
+	    echo; \
+	    echo 'Wrong version of requirement: $(1)'; \
+	    echo; \
+	    echo 'Please have a look at "README" to ensure'; \
+	    echo 'that your system meets all requirements.'; \
+	    echo; \
+	    exit 1; \
+	fi
+
+endef
+check-requirements: .stamp-check-requirements.h
+.stamp-check-requirements.h: $(MAKEFILE)
+	@echo '[check requirements]'
+	$(foreach REQUIREMENT,$(REQUIREMENTS),$(call CHECK_REQUIREMENT,$(REQUIREMENT)))
+	$(call CHECK_REQUIREMENT_VERSION,vagrant,[2-9]\.[0-9]\.[0-9])
```

Comment:
fwiw Ubuntu 16.04 LTS fails this check, with 1.8.1.

Reply:
Vagrant 2.0+ is required.

Reply:
To be clear, yes Ubuntu 16.04 LTS fails this check. That’s why the check exists :) I updated the docs to have directions for installing from the source.

Reply:
To be clear, yes Ubuntu 16.04 LTS fails this check. That’s why the check exists :) I updated the docs to have directions for installing from the source.